v2/concordium/protocol-level-tokens.proto

```diff
@@ -0,0 +1,144 @@
+syntax = "proto3";
+
+package concordium.v2.plt;
+
+// This specifies the package we want to use for our generated Go code.
+// Has no effect on code generated on other languages.
+option go_package = "./pb";
+
+// This specifies the package we want to use for our generated Java classes.
+// Has no effect on code generated on other languages.
+option java_package = "com.concordium.grpc.v2.plt";
+
+// This specifies that separate .java files will be generated for each of the Java classes/enums/etc. generated for the top-level messages, services, and enumerations,
+// and the wrapper Java class generated for this .proto file won't contain any nested classes/enums/etc.
+// If not generating Java code, this option has no effect.
+option java_multiple_files = true;
+
+// This specifies the package we want to use for our generated C# classes.
+// Has no effect on code generated on other languages.
+option csharp_namespace = "Concordium.Grpc.V2.Plt";
+
+import "kernel.proto";
+
+// The unique symbol and identifier of a protocol level token.
+message TokenId {
+  // Unique identifier for the token, guaranteed to be distinct across the
+  // entire concordium blockchain.
+  string symbol = 1;
+}
+
+// PLT amount representation.
+message TokenAmount {
+  // Significant figures 
+  uint64 figures = 1;
+  // Number of decimals in the representation
+  uint32 scale = 2;
+}
+
+// Token state at the block level
+message TokenState {
+  // The token name. This should be unique, but uniqueness is enforced only by
+  // the chain governance, not by the consensus protocol.
+  string token_name = 1;
+  // The reference of the module implementing this token.
+  concordium.v2.ModuleRef token_module_ref = 2;
+  // A URL pointing to additional meta information about the token.
+  optional string token_meta_url = 3;
+  // Account address of the issuer. The issuer is the holder of the nominated
+  // account which can perform token-governance operations.
+  concordium.v2.AccountAddress issuer = 4;
+  // Feature flags with the following semantics by position:
+  // 0: Whether the token supports an allow list.
+  // 1: Whether the token supports a deny list.
+  // 2: Whether the token supports minting.
+  // 3: Whether the token supports burning.
+  repeated bool feature_flag = 5;
+  // The total available token supply.
+  TokenAmount total_supply = 6;
```

Comment:
I suggest adding `number of decimals` in the `TokenState` if that is not part of the metadata file.

Reply:
it is part of TokenAmount

Reply:
I think it's better to have the number of decimals in the `TokenAmount` together with the digits. Otherwise you will have to look up the decimals in another part of the state to compute the actual amount.

Reply:
True, didn't realize that the decimals are part of the `TokenAmount`.